internal/dkg/store.go

```diff
@@ -225,3 +225,14 @@ func (s *boltStore) MigrateFromGroupfile(beaconID string, groupFile *key.Group,
 
 	return s.SaveFinished(beaconID, &state)
 }
+
+func (s *BoltStore) NukeState(beaconID string) error {
```

Comment:
Why not lock it during the process here?

Reply:
taking a second look at the code, perhaps it can be entirely removed - it maintains a fake transaction between two writes on `SaveFinished` but in fact that is already supported out-of-the-box with bolt